app/controllers/GuidanceController.scala

```diff
@@ -31,20 +33,31 @@ import scala.concurrent.ExecutionContext.Implicits.global
 @Singleton
 class GuidanceController @Inject() (
     errorHandler: ErrorHandler,
-    view: render_page,
+    standardView: standard_page,
+    questionView: question_page,
+    formProvider: NextPageFormProvider,
     service: GuidanceService,
     mcc: MessagesControllerComponents
 ) extends FrontendController(mcc)
     with I18nSupport {
   val logger: Logger = Logger( getClass )
 
-  def getPage(path: String, questionPageUrl: Option[String] = None): Action[AnyContent] = Action.async { implicit request =>
-    val url = "/" + questionPageUrl.fold[String](path)(url => url.drop("/guidance/".length))
-    request.session.get(SessionKeys.sessionId).fold(Future.successful(NotFound(errorHandler.notFoundTemplate)))( sessionId =>
-      service.getPage(url, sessionId).map {
-        case Some(page) => Ok(view(page))
-        case _ => NotFound(errorHandler.notFoundTemplate)
-      }
+  def getPage(path: String): Action[AnyContent] = Action.async { implicit request =>
+    sessionPage(s"/$path").map {
+        case Some(page: StandardPage) => Ok(standardView(page))
+        case Some(page: QuestionPage) => Ok(questionView(page, questionName(path), formProvider()))
+        case None => NotFound(errorHandler.notFoundTemplate)
```

Comment:
It might be more appropriate to return bad request here. We shouldn't be getting requests to this Url that haven't got session ids.

Reply:
Ok, will log and return BadRequest

Reply:
We will raise a ticket to improve error handling.